File: APTProject/app/src/main/java/com/demo/apt/MainActivity.java
package com.demo.apt;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import com.demo.annotation.BindView;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.tv_content)
    TextView tvContent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnifeUtils.bind(this);
        if (tvContent!=null){
            tvContent.setText("Hello APT");
        }

    }
}

File: AspectjProject/app/src/main/java/com/demo/aspectj/MainActivity.kt
package com.demo.aspectj

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


    }

    var index = 0

    @SingleClick
    fun onClick(view: View) {
        index++
        (view as TextView).text = "点击：$index"
    }
}